js/src/Figure.ts

```diff
@@ -100,14 +100,18 @@ export class Figure extends DOMWidgetView {
   }
 
   render() {
-    // we cannot use Promise.all here, since this.layoutPromise is resolved, and will be overwritten later on
-    this.displayed.then(() => {
-      // make sure we render after all layouts styles are set, since they can affect the size
-      this.layoutPromise.then(this.renderImpl.bind(this));
-    });
+    // We don't do anything, we'll wait for the widget to actually be visible
   }
 
   protected async renderImpl() {
+    if (!(this.attached && this.shown) || this.rendered) {
```

Comment:
```suggestion
    if (!this.attached || !this.shown || this.rendered) {
```
equivalent using demorgan's law, and easier on my brain :)